Frends.Salesforce.ExecuteQuery/Frends.Salesforce.ExecuteQuery/ExecuteQuery.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Newtonsoft.Json;
+using RestSharp;
+using System;
+using System.ComponentModel;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Frends.Salesforce.ExecuteQuery
+{
+    /// <summary>
+    /// Tasks class.
+    /// </summary>
+    public class Salesforce
+    {
+        /// <summary>
+        /// Execute a query to salesforce
+        /// [Documentation](https://github.com/FrendsPlatform/Frends.Salesforce/tree/main/Frends.Salesforce.ExecuteQuery)
+        /// </summary>
+        /// <returns></returns>
+        public static async Task<Result> ExecuteQuery(
+            [PropertyTab] Input input,
+            [PropertyTab] Options options,
+            CancellationToken cancellationToken
+        )
+        {
+            var query = WebUtility.UrlEncode(input.Query);
+            var client = new RestClient(input.Domain + "/services/data/v54.0/query/?q=" + query);
+            var request = new RestRequest("/", Method.Get);
+            string accessToken = "";
+
+            if (options.AuthenticationMethod is AuthenticationMethod.AccessToken)
+            {
+                if (string.IsNullOrWhiteSpace(options.AccessToken)) throw new ArgumentException("Access token cannot be null when using Access Token authentication method");
+                request.AddHeader("Authorization", "Bearer " + options.AccessToken);
+            }
+
+            if (options.AuthenticationMethod is AuthenticationMethod.OAuth2WithPassword)
+            {
+                accessToken = await GetAccessToken(options.ClientID, options.ClientSecret, options.Username, options.Password + options.SecurityToken, cancellationToken);
+                request.AddHeader("Authorization", "Bearer " + accessToken);
+            }
+
+            var response = await client.ExecuteAsync(request, cancellationToken);
+
+            if (options.AuthenticationMethod is AuthenticationMethod.OAuth2WithPassword && options.ReturnAccessToken)
+                return new ResultWithToken { Body = JsonConvert.DeserializeObject<dynamic>(response.Content), RequestIsSuccessful = response.IsSuccessful, ErrorException = response.ErrorException, ErrorMessage = response.ErrorMessage, Token = accessToken };
+            else
+                return new Result { Body = JsonConvert.DeserializeObject<dynamic>(response.Content), RequestIsSuccessful = response.IsSuccessful, ErrorException = response.ErrorException, ErrorMessage = response.ErrorMessage };
+        }
+
+        #region HelperMethods
+
+        /// <summary>
+        /// Get OAuth2 access token.
+        /// This method is public since it is used also in Unit tests.
+        /// </summary>
+        public static async Task<string> GetAccessToken(string clientId, string clientSecret, string username, string passwordWithSecurityToken, CancellationToken cancellationToken)
+        {
+            var authClient = new RestClient(@"https://login.salesforce.com/services/oauth2/token");

```

Comment:
We need to put this URL to options at least, because this is not the same for Salesforce sandbox environments (which are always used for dev/test purposes). The sandboxes use the "https://test.salesforce.com/services/oauth2/token" for getting tokens. So either just param the whole URL or maybe even better - have a checkbox that makes this task go to test URL.

Reply:
Moved to options.